akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -87,21 +87,7 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
         .foldLeft(Map.empty[AbstractServiceKey, Set[ActorRef[_]]])(changesForKey(_, _))
     }
 
-    val adapter: ActorRef[Replicator.ReplicatorMessage] =
-      ctx.spawnAdapter[Replicator.ReplicatorMessage] { (x: Replicator.ReplicatorMessage) ⇒
-        x match {
-          case changed @ Replicator.Changed(ReceptionistKey) ⇒
-            val value = changed.get(ReceptionistKey)
-            val oldState = state
-            state = ServiceRegistry(value) // is that thread-safe?
```

Comment:
Seems I had a hunch when I wrote that code ;)

IIUC there's no good reason why `spawnAdapter` could not work in a way that this would still be thread-safe, right? But it probably doesn't matter if we are revamping / removing `spawnAdapter` anyway.